crypto/pem/pem_test.cc

```diff
@@ -43,3 +58,141 @@ TEST(PEMTest, NoRC4) {
+
+using TmpFilePtr = std::unique_ptr<FILE, TmpFileDeleter>;
+
+static void* d2i_ASN1_INTEGER_void(void ** out, const unsigned char **inp, long len) {
+  return d2i_ASN1_INTEGER((ASN1_INTEGER **)out, inp, len);
+}
+
+static int i2d_ASN1_INTEGER_void(const void *a, unsigned char **out) {
+  return i2d_ASN1_INTEGER((ASN1_INTEGER *)a, out);
+}
+
+TEST(PEMTest, WriteReadASN1IntegerPem) {
+#if defined(OPENSSL_ANDROID)
+  // On Android, when running from an APK, |tmpfile| does not work. See
+  // b/36991167#comment8.
+  GTEST_SKIP();
+#endif
+  // Numbers for testing
+  long nums[8] = {
+      0x00000001L,
+      0x00000100L,
+      0x00010000L,
+      0x01000000L,
+      -2L};
+
+  for(size_t i = 0; i < 8; i++) {
+    long original_value = nums[i];
+
+    // Create an ASN1_INTEGER with value 42
+    bssl::UniquePtr<ASN1_INTEGER> asn1_int(ASN1_INTEGER_new());
+    ASSERT_TRUE(asn1_int.get());
+    ASSERT_TRUE(ASN1_INTEGER_set(asn1_int.get(), original_value));
+
+    // Create buffer for writing
+    TmpFilePtr pem_file(tmpfile());
+    ASSERT_TRUE(pem_file.get());
+
+    // Write the ASN1_INTEGER to a PEM-formatted string
+    ASSERT_TRUE(PEM_ASN1_write(i2d_ASN1_INTEGER_void, "ASN1 INTEGER",
+                               pem_file.get(), asn1_int.get(), nullptr, nullptr,
+                               0, nullptr, nullptr));
+
+    rewind(pem_file.get());
+    // Read the ASN1_INTEGER back from the PEM-formatted string
+    bssl::UniquePtr<ASN1_INTEGER> read_integer((ASN1_INTEGER *)PEM_ASN1_read(
+        d2i_ASN1_INTEGER_void, "ASN1 INTEGER", pem_file.get(), nullptr,
+        nullptr, nullptr));
+    ASSERT_TRUE(read_integer);
+
+    // Check if the read ASN1_INTEGER has the same value as the original
+    long read_value = ASN1_INTEGER_get(read_integer.get());
+    ASSERT_EQ(original_value, read_value);
+  }
+}
+
+const char* kPemRsaPrivateKey = "-----BEGIN ENCRYPTED PRIVATE KEY-----\n"
+    "MIHsMFcGCSqGSIb3DQEFDTBKMCkGCSqGSIb3DQEFDDAcBAhz3vU103jx3wICCAAw\n"
+    "DAYIKoZIhvcNAgkFADAdBglghkgBZQMEASoEEA6vMhRLgHZuHFa+eiecYCgEgZDB\n"
+    "E8EOzjGQuu4D0TVAjOa3Peb9/MzQz3t09m5pvNBFKrEl96gefpZdni5qQk34ukj9\n"
+    "/kryXymP72m4Ch7vbmew08x5x9L69BpfsQLF1yyvAJVtEZ0a1Zqcn5veuoH2WtJT\n"
+    "ZTrZtc5Eb+tAjMVzRXPZ80cUwCbbpb9KHUX8spwtG10I1VxJ18X31FVpGORdr0A=\n"
+    "-----END ENCRYPTED PRIVATE KEY-----";
+
+
+TEST(PEMTest, ReadPrivateKeyPem) {
+  bssl::UniquePtr<BIO> read_bio(BIO_new_mem_buf(kPemRsaPrivateKey, strlen(kPemRsaPrivateKey)) );
+  ASSERT_TRUE(read_bio.get());
+  bssl::UniquePtr<EC_KEY> ec_key(PEM_read_bio_ECPrivateKey(read_bio.get(), nullptr, pem_password_callback, (void*)SECRET));
+  ASSERT_TRUE(ec_key);
+  const EC_GROUP* p256 = EC_GROUP_new_by_curve_name(NID_X9_62_prime256v1);
+  ASSERT_EQ(p256, EC_KEY_get0_group(ec_key.get()));
+}
+
+
+TEST(PEMTest, WriteReadRSAPem) {
+  bssl::UniquePtr<RSA> rsa(RSA_new());
+  ASSERT_TRUE(rsa);
+  bssl::UniquePtr<BIGNUM> bn(BN_new());
+  ASSERT_TRUE(bn);
+  BN_set_u64(bn.get(), RSA_F4);
+#ifdef BORINGSSL_FIPS
```

Comment:
a minor nit, but just to be consistent with how we've been consuming this:
 ```suggestion
#if defined(BORINGSSL_FIPS)
```